vision.md

```diff
@@ -206,26 +206,188 @@ The framework includes modules for cloud connectivity, data mapping, device
 management, intra-edge communication, and certificate management, all aspects
 and challenges our target personas need to address.
 
+We therefore define the following requirements.
+
+### Non-functional requirements
+
+<!--
+The following list of non-functional requirements should be considered
+non-final. It might be extended in the future.
+-->
+
+1. Multi-paradigm edge framework
+    - Several kinds of users are be able to consume thin-edge.io at different
+      stages (e.g. using the Rust code as a framework to develop their own
+      solutions, using (networking)interfaces like for example MQTT to attach
+      their own infrastructure to thin-edge.io, or using the final
+      deployment-ready binary to solve their problems).
+1. Introspective functionality to enable quality assurance as well as support of
+   deployments
+    - Knowing what happens inside a system is crucial to be able to quickly
+      identify misbehaving components. Users are be able to access an overview
+      of the communication of the different components
+1. Ability to attach data sources and data sinks to thin-edge.io via different
+   mechanisms, either by implementing connectors directly in the framework using
+   rust code, attaching them via some networking mechanism (e.g. MQTT) or other
+   mechanisms the framework provides
+    1. The framework ships with connectivity functionality for the most common
+       connectivity requirements
+    1. Configuration of connectivity functionality is exposed to the user
+    1. Connectivity over the same mechanism but with different configuration
+       settings is easily doable for the user (e.g. using MQTT to connect to
+       different brokers with completely different settings)
+1. Easy deployment and little friction when using thin-edge.io as a framework to
+   develop their own solutions based on it
+1. Trust in the user to do the right thing with the tools provided, respect
+   their decisions and guide them towards a maintainable solution by design
+    - Users are assumed to be experts in their domain and should be given the
+      tools they need to solve the problems they have. Unnecessarily burdening
+      them with abstractions only causes frustration
+1. High configurability and tweakability with high trust in custom changes
+   without having to redefine the world on small changes
+1. Allow precise configuration inside common solutions shipped by the framework
+    - Users are be able to change a small detail of their deployment without
+      having to then define unrelated elements
+    - As an example: Users wanting to set an SSL certificate should not need to
+      worry about also defining their DNS setup.
```

Comment:
I don't see this example as relevant in the context of IoT where a device certificate is not related to host names.

Reply:
That is the point! Them not being related is exactly why that should not happen.